lib/vsc/utils/exceptions.py

```diff
@@ -81,6 +84,28 @@ def __init__(self, msg, *args, **kwargs):
         if args:
             msg = msg % args
 
+        if self.LOC_INFO_TOP_PKG_NAMES is not None:
+            # determine correct frame to fetch location information from
+            frames_up = 1
+            if self.__class__ != LoggedException:
+                # move a level up when this instance is derived from LoggedException
+                frames_up += 1
+
+            # figure out where error was raised from
+            # current frame: this constructor, one frame above: location where this LoggedException was created/raised
+            frameinfo = inspect.getouterframes(inspect.currentframe())[frames_up]
+
+            # determine short location of Python module where error was raised from,
+            # i.e. starting with an entry from LOC_INFO_TOP_PKG_NAMES
+            path_parts = frameinfo[1].split(os.path.sep)
+            relpath = path_parts.pop()
+            while path_parts and not any([relpath.startswith(n + os.path.sep) for n in self.LOC_INFO_TOP_PKG_NAMES]):
```

Comment:
copy-pasting from reply to remark, which is hidden due to the updates:

no, it's not always false, since `relpath` is updated via `os.path.join` in the body of the while loop

I'm using the `os.path.sep` as an 'end of name' marker to make sure I don't have a partial match, e.g.:

```python
'easybuild/tools/filetools.py'.startswith('easybuild/') == True
'easybuild-framework/easybuild/tools/filetools.py'.startswith('easybuild/') == False
```

What I'm doing here is trimming something like `/home/kehoste/easybuild-framework/easybuild/tools/filetools.py` down to just `easybuild/tools/filetools.py`

I agree this looks rather strange, but I haven't found a better to do this.

`path_parts` is a list like `['', 'home', 'kehoste', 'easybuild-framework', 'easybuild', 'tools', 'filetools.py']`, so I'm taking the end of the list until I have something that starts with `easybuid/` (or `vsc/`)